health/health.go

```diff
@@ -170,6 +170,20 @@ func Register(name string, check Checker) {
 	registeredChecks[name] = check
 }
 
+// Unregister removes the named checker.
+func Unregister(name string) {
```

Comment:
I suppose it would be possible to skip the registration and just call the checker directly, but if we did that, the unit test would be testing something very different from what the registry actually does. It wouldn't check interpretation of the config structure or registration of the checks, only the fact that the health package's HTTP check or file check works as promised.

If we do register the checks, we need some way to remove registrations, so subsequent tests don't interfere with each other. Without `UnregisterAll`, there's a panic when the second test tries to call `RegisterHealthChecks`.

It would be possible to combine these two tests into one, but that feels less flexible and less robust. If we ever end up in a situation where another test registers a health check, we'd have the same problem.

I'm open to suggestions on how to deal with this.